src/pkg/scan/report/manager.go

```diff
@@ -79,34 +74,134 @@ type Manager interface {
 	//  Returns:
 	//    []*scan.Report : report list
 	//    error          : non nil error if any errors occurred
-	GetBy(digest string, registrationUUID string, mimeTypes []string) ([]*scan.Report, error)
-
-	// Get the report for the given uuid.
-	//
-	//  Arguments:
-	//    uuid string  : uuid of the scan report
-	//
-	//  Returns:
-	//    *scan.Report : scan report
-	//    error        : non nil error if any errors occurred
-	Get(uuid string) (*scan.Report, error)
+	GetBy(ctx context.Context, digest string, registrationUUID string, mimeTypes []string) ([]*scan.Report, error)
 
 	// Delete the reports related with the specified digests (one or more...)
 	//
 	//  Arguments:
+	//    ctx context.Context : the context for this method
 	//    digests ...string : specify one or more digests whose reports will be deleted
 	//
 	//  Returns:
 	//    error        : non nil error if any errors occurred
-	DeleteByDigests(digests ...string) error
+	DeleteByDigests(ctx context.Context, digests ...string) error
 
-	// GetStats retrieves and calculates the overall report stats organized by status targeting the
-	// given requester.
+	// List reports according to the query
+	//
 	//  Arguments:
-	//    requester string : the requester of the scan (all)
+	//    ctx context.Context : the context for this method
+	//    query *q.Query : the query to list the reports
 	//
 	//  Returns:
-	//    *all.AllStats: stats object including the related metric data
+	//    []*scan.Report : report list
 	//    error        : non nil error if any errors occurred
-	GetStats(requester string) (*all.Stats, error)
+	List(ctx context.Context, query *q.Query) ([]*scan.Report, error)
+}
+
+const (
+	reportTimeout = 1 * time.Hour
+)
+
+// basicManager is a default implementation of report manager.
+type basicManager struct {
+	dao scan.DAO
+}
+
+// NewManager news basic manager.
+func NewManager() Manager {
+	return &basicManager{
+		dao: scan.New(),
+	}
+}
+
+// Create ...
+func (bm *basicManager) Create(ctx context.Context, r *scan.Report) (string, error) {
+	// Validate report object
+	if r == nil {
+		return "", errors.New("nil scan report object")
+	}
+
+	if len(r.Digest) == 0 || len(r.RegistrationUUID) == 0 || len(r.MimeType) == 0 {
+		return "", errors.New("malformed scan report object")
+	}
+
+	r.UUID = uuid.New().String()
+
+	// Insert
+	if _, err := bm.dao.Create(ctx, r); err != nil {
+		return "", err
+	}
+
+	return r.UUID, nil
+}
+
+func (bm *basicManager) Delete(ctx context.Context, uuid string) error {
+	query := q.Query{Keywords: q.KeyWords{"uuid": uuid}}
+	count, err := bm.dao.DeleteMany(ctx, query)
```

Comment:
what about count >1?

Reply:
`count > 1` will never happen because `uuid` is unique in the table.